src/pages/Profiles.vue

```diff
@@ -486,6 +486,40 @@ export default class Profiles extends Vue {
         }
     }
 
+    async fixThunderstoreFile(file: string) {
+        let headerLoc = -1;
+
+        // zip end header starts with a 4 byte signature
+        const zipEndHeader = Buffer.from([0x50, 0x4b, 0x05, 0x06]);
+    
+        // stream until we find the end header
+        await fs.createReadStream(file, 8192, async (stream) => {
+            for await(const data of stream) {
+                const index = data.lastIndexOf(zipEndHeader);
+                if (index !== -1) {
+                    headerLoc += index;
+                    break;
+                } else {
+                    headerLoc += data.length;
+                }
+            }
+        });
+
+        // ensure exitst
+        if (headerLoc === -1) {
+            throw new Error("Failed to find zip end header");
```

Comment:
We tend to throw `R2Error` instead of `Error` as we use it for display purposes. They help the user identify what went wrong and how to potentially solve it. It's also somewhat an identifier as to where it broke in the process as the user will send a screenshot.